### ISSUE TITLE
Support MOI.Semicontinuous variable

### ISSUE THREAD

**Issue by mzy2240** [COPT-Public/COPT.jl#20]
When running `@constraint(model, [i = 1:n], x[i] in MOI.Semicontinuous(x_limit[1], x_limit[2]))` with COPT.Optimizer, I got this error `Constraints of type MathOptInterface.VariableIndex-in-MathOptInterface.Semicontinuous{Int64} are not supported by the solver.`. Gurobi.jl supports `MOI.Semicontinuous` out-of-box, so can we also add the support for this?

**Comment by odow** [comment]
Can you try:
```Julia
@constraint(model, [i = 1:n], x[i] in MOI.Semicontinuous{Float64}(x_limit[1], x_limit[2]))
```
JuMP should be able to automatically reformulate this.

**Comment by odow** [comment]
> That probably means we want to auto-promote/convert the type when passing to the function.

Yep. This is what I meant by "If it works, we should fix this in JuMP."

I've opened a JuMP issue: https://github.com/jump-dev/JuMP.jl/issues/3300
